docs/lambdas/README.md

```diff
@@ -0,0 +1,106 @@
+
+```console
+python3 -m venv env # Only needed the first time (env folder is in .gitignore)
+source env/bin/activate
+```
+
+You can then install project dependencies by running:
+
+```console
+pip install -r test_requirements.txt #
+```
+
+You should now be able to run your lambda locally with:
+
+```console
+python main.py # You may need to set the appropriate environment variables to pass permissions and configuration.
+```
+
+### Running tests
+
+We use [`tox`](https://tox.wiki/en/4.13.0/) and [`pytest`](https://docs.pytest.org/en/8.0.x/) to manage running tests, and you should find a `tox.ini` above your project directory.
+
+To run tests from the project directory, ensure you have the correct python version and environment set up then run:
+
+```console
+tox
+```
+
+### Adding dependencies
+
+We use `pip-compile` to generate transitive dependencies in a []`requirement.txt` file from direct dependencies in a `requirements.in` file](https://stackoverflow.com/questions/66751657/what-does-pip-compile-do-what-is-its-use-how-do-i-maintain-the-contents-of-my).
+
+New dependencies should be added to `requirements.in` and then the following commands can be run to update transitive dependencies. Ensure you have the correct python version and environment selected then:
+
+```console
+pip install pip-tools # Only needed the first time to install pip-compile
+pip-compile
+pip-compile test_requirement.in # updates test_requirement.txt, incorporating requirements.txt
+```
+
+## Deployment
+
+Our continuous integration environment is configured to build, upload and deploy AWS Lambda functions but this can also be done locally.
+
+In terraform we use the [`wellcomecollection/terraform-aws-lambda` module](https://github.com/wellcomecollection/terraform-aws-lambda) to provide a template for our lamdbas that provides some default configuration like [forwarding Lambda logs to Elasticsearch](https://github.com/wellcomecollection/elasticsearch-log-forwarder). This module expects you to upload zipped pacakages to S3 and then update the Lambda by referencing a newly uploaded package to deploy code changes.
+
+This repository has two scripts to help with this process:
+
+### `./builds/publish_lambda_zip.py`
+
+This script downloads all the dependencies specified by requirements.txt, and packages them alongside your application code in a `.zip` file in `.lambda_zip` at the root of the repository. It then uploads them to S3 in `s3://wellcomecollection-platform-infra/lambdas/project_name`. The script requires AWS credentials. You can run the script locally (ensure you have selected the Python 3.6 env using pyenv):
+
+```console
+# Get AWS credentials
+aws-azure-login 
+# Switch to the correct python version
+pyenv shell 3.6 
+# Build and publish project, specify location of src folder for your project
+AWS_PROFILE=platform-developer ./builds/publish_lambda_zip.py ebsco_adapter/ebsco_adapter
+```
```

Comment:
These instructions should probably mention the new second "suffix" argument now.

Reply:
Oops, I thought i'd binned that as unnecessary. Hold on, i'm going to trim some commits off the end of this branch. The extra suffix isn't required.

Reply:
even better